### PR TITLE
Iteration 3 for file detail & new sharing

### PR DESCRIPTION
**Pull request by AndyScherzinger**
Iteration 1 can be found here: #2533 :ballot_box_with_check:
Iteration 2 can be found here: #2573 :ballot_box_with_check:

WiP for #2485 
* [x] https://github.com/nextcloud/android-library/pull/149 merged
* [x] change back dependency to `master-SNAPSHOT`as soon as lib PR has been merged
* [x] ability to properly configure email shares

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky @mario to be discussed if this is the right approach. Atm my change would simply store a flag in the database if a share has a password but wouldn't store the password. This is imho the way to go while of course we could also store the password I simply wouldn't do that but have the user input a new password. 
This is imho also how the webUI behaves, thus you can always change the password but you cannot recover the actual one.

**Comment by tobiasKaminsky** [comment]
Totally fine with it. Storing a password is not needed as you can always set a new one.

**Comment by AndyScherzinger** [comment]
@mario @ardevd @tobiasKaminsky please review :)

File: src/main/java/com/owncloud/android/operations/UpdateSharePermissionsOperation.java [hunk 1]
```diff
@@ -117,6 +135,8 @@ private void updateData(OCShare share) {
         } else {
             share.setIsFolder(false);
         }
+
+        share.setIsPasswordProtected((mPassword != null && mPassword.length() > 0));
```
Comment (mario) [inline src/main/java/com/owncloud/android/operations/UpdateSharePermissionsOperation.java]: TextUtils.isEmpty(mPassword)? :)

Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/operations/UpdateSharePermissionsOperation.java]: nice catch! :tada:

**Comment by mario** [comment]
One small comment.

**Comment by mario** [comment]
👍

File: src/main/java/com/owncloud/android/operations/UpdateSharePermissionsOperation.java (hunk 1 above)
Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/operations/UpdateSharePermissionsOperation.java]: Fixed via https://github.com/nextcloud/android/pull/2671/commits/723ec59020bcc44792b43a019dcaae20c96a8108

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky please review :)

**Comment by tobiasKaminsky** [comment]
Can you give me a quick guide how to share a file via email?
- open up file detail
- go to sharing
- enter email address:


(on webUI, I see "test123@test.de (email)

**Comment by AndyScherzinger** [comment]
You can't atm, but you ate able to change existing ones regarding permissions and password. That didn't work before.

For actually creating new mail shares we need to find out how to do that...

**Comment by tobiasKaminsky** [comment]
:+1: 
Editing existing email share is working :tada:

**Pull request merged**